spec/commands/write_batch_spec.cr

```diff
@@ -0,0 +1,52 @@
+require "../spec_helper"
+
+describe "WriteBatch" do
+  path = "tmp/db_write_batch"
+
+  describe "#delete_range" do
+      it "deletes range" do
+        db = RocksDB::DB.new(path)
+
+        db.put(Bytes[0, 0], Bytes[9])
+        db.put(Bytes[0, 1], Bytes[9])
+        db.put(Bytes[0, 2], Bytes[9])
+
+        batch = RocksDB::WriteBatch.new
+
+        key_prefix_start = Bytes[0, 0, 0, 0]
+        key_prefix_end = Bytes[255, 255, 255, 255]
+
+        batch.delete_range(key_prefix_start, key_prefix_end)
```

Comment:
In my environment, an error occurred when executing here.

```
WriteBatch
  #delete_range
    deletes range

Failures:

  1) WriteBatch #delete_range deletes range

Error running at_exit handler: Index out of bounds
```

```console
$ crystal --version
Crystal 1.11.1 [0aa30372c] (2024-01-11)

LLVM: 15.0.7
Default target: x86_64-unknown-linux-gnu
```

Reply:
Yeah I see that too. I don't know why it sometimes works. Do you have any idea what's going on?

Reply:
What version of rocksdb are you running?

Reply:
```console
$ apt-show-versions librocksdb-dev
librocksdb-dev:amd64/jammy 6.11.4-3 uptodate

$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.3 LTS"
```

Reply:
Ok that version is fairly old. I just upgraded to 8.11.0 and the iterator is now throwing an error. Do you want to enforce a rockdb version? Because we will have to support two versions.

Reply:
You haven't replied for a while, so I'm going to go ahead and update this library to work with the latest rocksdb (8.*+). This means changes to the iterator and fixing the batch write.

Reply:
If we can support rocksdb8, we would only need to support its latest version.

However, I feel that it would probably require a great deal of corrective work.
PR is welcome. 😃

Reply:
ok I'll take a look at it.